agent/util-scripts/pbench-tool-data-sink

```diff
@@ -419,11 +422,11 @@ class ToolDataSink(Bottle):
                 persistent_tools = []
                 transient_tools = []
                 for tool_name in tools.keys():
-                    if tool_name in PromCollector.allowed_tools:
+                    if tool_name in self.tool_metadata.getPersistentTools():
                         persistent_tools.append(tool_name)
                     elif tool_name in BaseCollector.allowed_tools:
                         noop_tools.append(tool_name)
-                    else:
+                    elif tool_name in self.tool_metadata.getTransientTools():
```

Comment:
I like the extra validation of the `elif` instead of the old `else` ... but this raises a possible error case where `tool_name` is something else. We're no longer defaulting that to "transient" as we did before. Not entirely sure we should care since it's an unlikely error; but it might be worth at least a warning log entry "just in case".

Reply:
@portante, not really sure what you mean by that. I don't see any relevant errors. We're just looping through the selected tools from Redis, and trying to categorize each into persistent/transient. There should never be any selected tools that aren't in the list. However, if there are, the "old" code (in an earlier commit) silently categorized them as transient; the new code won't categorize the tool at all. Which for example means that a "missing" sole transient tool would trigger the `elif not transient_tools:` below and won't expect an upload to the sink.

I like the `elif` rather than the `else` but if it does somehow happen through some weird skew or error, it'd be nice to see some indication, either via a log or maybe even an assert.